spec/lib/expansion_rules_spec.rb

```diff
@@ -38,6 +38,70 @@
     end
   end
 
+  describe ".find_custom_expansion_fields" do
+    before do
+      stub_const(
+        "ExpansionRules::CUSTOM_EXPANSION_FIELDS",
+        [
+          { document_type: :news, fields: %i[news_a] },
+          { document_type: :news,
+            link_type: :breaking_news,
+            fields: %i[news_b news_c] },
+          { document_type: :news,
+            link_type: :local_news,
+            fields: %i[news_d news_e] },
+          { document_type: :feature, fields: %i[feature_a feature_b] },
+          { document_type: :editorial,
+            link_type: :current_events,
+            fields: %i[editorial_a] },
+          { document_type: :editorial,
+            link_type: :world_events,
+            fields: %i[editorial_b] },
+        ],
+      )
+    end
+
+    it "returns nil when document type doesn't match" do
+      expect(rules.find_custom_expansion_fields(:event)).to be_nil
+    end
+
+    it "can accept strings instead of symbols" do
+      strings = rules.find_custom_expansion_fields("news", "local_news")
+      symbols = rules.find_custom_expansion_fields(:news, :local_news)
+      expect(strings).to eq(symbols)
+    end
+
+    context "when providing a document_type and a link" do
+      it "returns the fields for a link that matches" do
+        expect(rules.find_custom_expansion_fields(:news, :breaking_news))
+          .to eq(%i[news_b news_c])
+      end
+
+      it "returns the fields of the document type if that has no link type defined" do
+        expect(rules.find_custom_expansion_fields(:news, :other_link))
+          .to eq(%i[news_a])
+      end
+    end
+
+    context "when providing a document_type without a link" do
+      it "returns the fields for a document_type without links defined" do
+        expect(rules.find_custom_expansion_fields(:feature, nil))
+          .to eq(%i[feature_a feature_b])
+      end
+
+      it "collates the fields for document_types with multiple link types" do
+        expect(rules.find_custom_expansion_fields(:news, nil))
+          .to match_array(%i[news_a news_b news_c news_d news_e])
+      end
+
+      it "includes default fields for a document type that only exists in "\
+        "custom fields with a link" do
+        expect(rules.find_custom_expansion_fields(:editorial, nil))
+          .to contain_exactly(:editorial_a, :editorial_b, *ExpansionRules::DEFAULT_FIELDS)
```

Comment:
I'm a little unsure why we're using `eq` on line 89, `match_array` on line 94 and `contain_exactly` on line 100.

Reply:
Thanks for the explanation and tweak.